x-pack/metricbeat/module/containerd/helper.go

```diff
@@ -0,0 +1,28 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package containerd
+
+import "github.com/elastic/beats/v7/libbeat/common"
+
+// SetCIDandNamespace sets container.id ECS field and containerd.namespace module field
+func SetCIDandNamespace(event common.MapStr) (common.MapStr, common.MapStr, string) {
```

Comment:
I would split it into 2 functions like `GetAndDeleteCid` and `GetAndDeleteNamespace`,  which would return the id and namespace explicitly. Current implementation gets an Event and returns `moduleFields`, `rootFields` and cID, so it's a mix and not a straightforward in<->out function.  Also "set" naming implies that the function sets sth on an already existent object, var or whatever.

Reply:
Ok!

Reply:
5be9960ff9482e9b4b736f3fd742817293cf7c04